tests/e2e/client/apiproxy_bookstore_quota_test.py

```diff
@@ -45,29 +45,30 @@ def verify_quota_control(self):
     FLAGS.verbose = False
     self.set_verbose(FLAGS.verbose)
 
-    # exhaust the quota in the current window.
-    print("Exhaust current quota...");
-    response = self._call_http(path='/quota_read',
-                               api_key=FLAGS.api_key)
-    if response.status_code != 429:
-      for i in range(10000):
-        response = self._call_http(path='/quota_read',
-                                   api_key=FLAGS.api_key)
+
+    def _exhaust_quota():
+      for i in range(100):
+        time.sleep(1)
+        try:
+          response = self._call_http(path='/quota_read',
+                                     api_key=FLAGS.api_key)
+        except Exception, e:
+          print "Exception {0} occurred".format(e)
+          continue
         if response.status_code == 429:
           break;
-        elif i == 9999:
+        elif i == 99:
           sys.exit(utils.red("Fail to exhaust quota"))
 
+    # exhaust the quota in the current window.
+    print("Exhaust current quota...")
```

Comment:
Could you add the comment on the algorithm you are using?  Like to know what are the purposes of these
 the exhaust_quota() and sleep(5) calls

Reply:
Done.